### DISCUSSION TITLE
subscriptions-transport-ws is still being recommended

### DISCUSSION THREAD

**Discussion by o-az** [urql-graphql/urql#3569] [Ideas]
[subscriptions-transport-ws](https://github.com/apollographql/subscriptions-transport-ws) has been archived for over a year but is still being recommended on [this page](https://commerce.nearform.com/open-source/urql/docs/advanced/subscriptions/):

> For backends supporting subscriptions-transport-ws, [Apollo's subscriptions-transport-ws package](https://github.com/apollographql/subscriptions-transport-ws) can be used.

Then under that:

> The subscriptions-transport-ws package isn't actively maintained. If your API supports the new protocol or you can swap the package out, consider using [graphql-ws](https://commerce.nearform.com/open-source/urql/docs/advanced/subscriptions/#setting-up-graphql-ws) instead.

Might be a good idea to not suggest it at all.

**Comment by kitten** [comment]
This isn't an active recommendation, to highlight this:
> For backends **supporting** graphql-ws
> For backends **supporting** subscriptions-transport-ws,

The WebSocket transports that are available for GraphQL are unfortunately not spec-based. This means that the specific transport-characterics of each library must be used.

For example, some GraphQL libraries for other languages have used the same transport pattern of either of these libraries to support WebSockets, so users can simply use one of the above libraries that's supported by them.
And as far as I know, some still use `subscriptions-transport-ws`.

This is why it's important to call out that users have to check which of the two their backends supports rather than to assume that they can use any. If you control and write your own GraphQL API in JS specifically then that doesn't apply to you, but we can't remove the section on `subscriptions-transport-ws` while APIs are still using it.

In practice, `fetch` is recommended, since it just works out of the box, but that also requires users to check whether their API of choice supports it: https://urql.dev/goto/docs/advanced/subscriptions/#using-fetch-for-subscriptions
